app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    history = {}
+
+    def wrapper(*args) -> Callable:
+        if isinstance(args, (int, float, str, tuple)):
+            if args in history:
+                print("Getting from cache")
+                return history[args]
+            else:
+                result = func(*args)
+                print("Calculating new result")
+                history[args] = result
+                return result
```

Comment:
you can use only one return after `if else` statement

Reply:
don't get this point
could you please explain it?